nixos/modules/services/networking/corerad.nix

```diff
@@ -4,18 +4,58 @@ with lib;
 
 let
   cfg = config.services.corerad;
+
+  writeTOML = name: x:
+    pkgs.runCommandNoCCLocal name {} ''
+      echo '${builtins.toJSON x}' | ${pkgs.go-toml}/bin/jsontoml > $out
+    '';
+
+  # Prefer settings attributes over a config file if both are set.
+  coreradTOML = if cfg.settings != null then
+    writeTOML "corerad.toml" cfg.settings
+  else
+    cfg.configFile;
+
 in {
-  meta = {
-    maintainers = with maintainers; [ mdlayher ];
-  };
+  meta.maintainers = with maintainers; [ mdlayher ];
 
   options.services.corerad = {
     enable = mkEnableOption "CoreRAD IPv6 NDP RA daemon";
 
+    settings = mkOption {
+      default = null;
+      type = types.attrs;
```

Comment:
https://github.com/NixOS/nixpkgs/pull/75584 (I intend to finish it soon and update the RFC) would contain a better type than `types.attrs`, which has problems with merging. To prevent those, you can turn off merging altogether with `type = types.uniq types.attrs`.

That PR will also contain a TOML generation function which then can be used.